src/functions-and-arrays.js

```diff
@@ -1,41 +1,126 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a,b) {
+  if (a > b) return a
+  else return b
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length === 0) {
+ 
+ return null;
+ 
+  }
+ 
+ let longestWord = "";
+ 
+ for (let i = 0; i < words.length; i++) {
+ 
+ if (words[i].length > longestWord.length) {
+ 
+ longestWord = words[i];
+      }
+   }
+ 
+  return longestWord;
+  }
+  
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(arrayOfNumbers) {
+  currSum = 0
+  arrayOfNumbers.forEach(number => {
+    currSum += number
+  })
+  return currSum
+}
+
 
 
+// Iteration #3.1 Bonus:    DOES NOT WORK
+function sum(mixedArr) {
+  if (mixedArr.length === 0) return 0
 
-// Iteration #3.1 Bonus:
-function sum() {}
+
+  let currSum = 0
+  mixedArr.forEach(value => {
+   if(typeof value === "number"){
+     currSum += value
+   }
+    else if (typeof value === "string"){
+      currSum += value.length
+    }
+    else if(typeof value === "boolean"){
```

Comment:
A boolean has a number value. false, when added to a number, it's like adding 0. true, is +1

Reply:
Therefore, you don't need the second argument, you can just do if (typeof value==="boolean"){currSum += value}

Reply:
You're only missing the last part, which entails the typeof being object (array's typeof is object), and that should throw an error, like throw new Error("Unsuported data type sir or ma'am")